woocommerce/woocommerce-android
# Issue/2844 order detail refactor

Fixes #2844. This is the feature branch for the order detail refactor to MVVM. All code in this PR has already been reviewed in previous PRs.

#### To Test
- Smoke testing the order detail screen to verify that the following cards display the correct information:
  - Display Order Status Info card. (PR: #2856 )
  - Display Shipping Lines Warning card, if order has more than 1 shipping methods. (PR: #2857 )
  - Display Customer Info card - Shipping/Billing address. (PR: #2860 )
  - Display Payment Info card. (PR: #2861 )
  - Display Order notes card. (PR: #2866 )
  - Display Refunds Info card. (PR: #2871 )
  - Display Product List card. (PR: #2872 )
  - Display Shipment Tracking list card, if WCS plugin is available. (PR: #2873 )
  - Display Shipping Labels card, if WCS plugin is available.  (PR: #2874 )
- Smoke the following order actions to verify that everything is working as expected.
  - Pull to refresh action on the Order Detail screen. (PR: #2875 )
  - Change Order Status (PR: #2875 )
  - Fulfill Order (PR: #2878 )
  - Issue Refund. (Open Issue Refund screen and update the Order Detail when refund is successful).
  - Open Refund detail screen. (PR: #2875 )
  - Send email using billing email. Call/message billing phone number. (PR: #2875 )
  - Add new order note. (Open Add order note screen and update the Order Detail when successful). (PR: #2876 )
  - Request Refund for a shipping label. (PR: #2876 )
  - Add Shipment Tracking for an order. (PR: #2877 )
  - Delete Shipment Tracking for an order. (PR: #2877 )
  - Track/Copy Tracking number for an order (with WCS or Shipment Tracking plugin enabled). (PR: #2877 )
- Run `OrderDetailViewModelTest`.
- Click on a new order notification to verify that it is redirecting to the Order detail screen.

Update release notes:
